src/database-layer/db.ts

```diff
@@ -354,23 +354,22 @@ export abstract class Tx {
 		}
 	}
 	public async rollback(): Promise<void> {
-		try {
-			const promise = this._rollback();
-			this.closeTransaction('Transaction has been rolled back.');
+		const promise = this._rollback();
+		this.closeTransaction('Transaction has been rolled back.');
 
-			await promise;
-		} finally {
-			this.listeners.rollback.forEach(asyncTryFn);
-			this.on = onRollback;
-			this.clearListeners();
-		}
+		this.listeners.rollback.forEach(asyncTryFn);
+		this.on = onRollback;
+		this.clearListeners();
+
+		await promise;
```

Comment:
Did you change the order of execution on purpose, now executing `await promise;` after what was originally in the `finally`?

Reply:
These changes snuck in by accident - I've removed them now. Thanks for catching